### DISCUSSION TITLE
Nx for shared components/codes across multiple repo

### DISCUSSION THREAD

**Discussion by MrIDK22** [nrwl/nx#10677] [General]
Hi all,

I came across Nx through Lerna's GitHub. Lerna sounded like what we need, and from the docs, Nx as well, but I am still not entirely sure if it fits the bill. My apologies if I misunderstood the target users for it, but here is a bit of a background story...

My team currently has several different web apps (different git repositories, not a single repo) written in React. These apps were originally started by different teams but are now left to us to enhance and maintain. As you might have expected, there are quite a few components and code (e.g. hooks, helpers, etc) that are similar and have highly duplicated code. We are currently looking for ways to get rid of duplicate code and make things consistent across all apps.

What my team is hoping for is:

1. To be able to refactor out the duplicated code and/or create new shared components/code to reuse across all applications.
2. Different web applications can and will need different versions of the components. E.g. App A uses component version 1.1, App B uses component version 1.2.
3. Components and/or pieces of code to be reused should ideally be version controlled independently. 
4. The ability to quickly change and test shared code in dev alongside the application using it instead of having to build and release them first before testing it in dev.
5. The ability to work with create-react-app without having to eject and customize the webpack config.

So... am I looking at the right tool here?

**Comment by MrIDK22** [comment]
Shameless bump...

**Reply by rebelnerd** [reply]
Did you ever get a reply to this?
